src/libraries/System.IO.FileSystem/tests/File/EncryptDecrypt.Windows.cs

```diff
@@ -0,0 +1,85 @@
+using System.Diagnostics;
+using System.Diagnostics.Eventing.Reader;
+using System.Security;
+using System.ServiceProcess;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.IO.Tests
+{
+    public partial class EncryptDecrypt
+    {
+        partial void LogEFSDiagnostics()
+        {
+            try
+            {
+                using var sc = new ServiceController("EFS");
+                _output.WriteLine($"EFS service is: {sc.Status}");
+                if (sc.Status != ServiceControllerStatus.Running)
+                {
+                    _output.WriteLine("Trying to start EFS service");
+                    sc.Start();
+                    _output.WriteLine($"EFS service is now: {sc.Status}");
+                }
+            }
+            catch (Exception e)
+            {
+                _output.WriteLine(e.ToString());
+            }
+
+            var hours = 1; // how many hours to look backwards
+            var query = @$"
+                        <QueryList>
+                          <Query Id='0' Path='System'>
+                            <Select Path='System'>
+                                *[System[Provider/@Name='Server']]
+                            </Select>
+                            <Select Path='System'>
+                                *[System[Provider/@Name='Service Control Manager']]
+                            </Select>
+                            <Select Path='System'>
+                                *[System[Provider/@Name='Microsoft-Windows-EFS']]
+                            </Select>
+                            <Suppress Path='System'>
+                                *[System[TimeCreated[timediff(@SystemTime) &gt;= {hours * 60 * 60 * 1000L}]]]
+                            </Suppress>
+                          </Query>
+                        </QueryList> ";
+
+            var eventQuery = new EventLogQuery("System", PathType.LogName, query);
+
+            var eventReader = new EventLogReader(eventQuery);
+
+            EventRecord record = eventReader.ReadEvent();
+            var garbage = new string[] { "Background Intelligent", "Intel", "Defender", "Intune", "BITS", "NetBT"};
+
+            _output.WriteLine("=====  Dumping recent relevant events: =====");
+            while (record != null)
+            {
+                string description = "";
+                try
+                {
+                    description = record.FormatDescription();
+                }
+                catch (EventLogException) { }
+
+                foreach (string term in garbage)
+                {
+                    if (description.Contains(term, StringComparison.OrdinalIgnoreCase))
+                        goto next;
+                }
+
+                _output.WriteLine($"{record.TimeCreated} {record.ProviderName} [{record.LevelDisplayName} {record.Id}] {description.Replace("\r\n", "  ")}");
+
+            next:
+                record = eventReader.ReadEvent();
+            }
```

Comment:
nit: I think that we can avoid using `goto` here:

```suggestion
            var garbage = new string[] { "Background Intelligent", "Intel", "Defender", "Intune", "BITS", "NetBT"};

            _output.WriteLine("=====  Dumping recent relevant events: =====");
            
            EventRecord? record = null;
            while ((record = eventReader.ReadEvent()) != null)
            {
                string description = "";
                try
                {
                    description = record.FormatDescription();
                }
                catch (EventLogException) { }

                if (description == "" || !garbage.Any(term => description.Contains(term, StringComparison.OrdinalIgnoreCase))
                {
                    _output.WriteLine($"{record.TimeCreated} {record.ProviderName} [{record.LevelDisplayName} {record.Id}] {description.Replace("\r\n", "  ")}");
                }
            }
```